File: sedmica 3/FIT_Api_Examples/FIT_Api_Example/Endpoints/Student/GetAll/StudentGetAllEndpoint.cs
using FIT_Api_Example.Data;
using FIT_Api_Example.Endpoints.Student.Pretraga;
using FIT_Api_Example.Helper;
using Microsoft.AspNetCore.Mvc;
using Microsoft.EntityFrameworkCore;

namespace FIT_Api_Example.Endpoints.Student.GetAll
{
    [Route("student-getall")]
    public class StudentGetAllEndpoint: MyBaseEndpoint<StudentGetAllRequest,  StudentGetAllResponse>
    {
        private readonly ApplicationDbContext _applicationDbContext;

        public StudentGetAllEndpoint(ApplicationDbContext applicationDbContext)
        {
            _applicationDbContext = applicationDbContext;
        }

        [HttpGet]
        public override async Task<StudentGetAllResponse> Obradi([FromQuery] StudentGetAllRequest request)
        {
            var student = await _applicationDbContext.Student
                .OrderByDescending(x => x.ID)
                .Select(x=>new StudentGetAllResponseStudent()
                {
                    ID = x.ID,
                    DatumRodjenja = x.DatumRodjenja,
                    Ime = x.Ime,
                    Prezime =  x.Prezime,
                    KorisnickoIme = x.KorisnickoIme,
                    OpstinaRodjenjaDrzava = x.OpstinaRodjenja.drzava.Naziv,
                    OpstinaRodjenjaNaziv = x.OpstinaRodjenja.description,
                    SlikaKorisnika = x.SlikaKorisnika
                })
                .ToListAsync();

            return new StudentGetAllResponse
            {
                Studenti = student
            };
        }
    }
}

File: sedmica 3/front-end/src/app/endpoints/student-endpoints/student-getall.endpoint.ts
import {MyBaseEndpoint} from "../MyBaseEndpoint";
import {MojConfig} from "../../moj-config";
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {Injectable} from "@angular/core";
@Injectable({providedIn: 'root'})
export class StudentGetAllEndpoint implements  MyBaseEndpoint<void, Student6PretragaResponse>{
  constructor(public httpClient:HttpClient) { }

  obradi(request: void): Observable<Student6PretragaResponse> {
    let url=MojConfig.adresa_servera+`/student/pretraga`;
      return this.httpClient.get<Student6PretragaResponse>(url, MojConfig.get_http_opcije());
    }
}

export interface Student6PretragaResponse {
  studenti: Student6PretragaResponseStudenti[]
}

export interface Student6PretragaResponseStudenti {
  id: number
  ime: string
  prezime: string
  opstinaRodjenjaNaziv: string
  opstinaRodjenjaDrzava: string
  datumRodjenja: string
  korisnickoIme: string
  slikaKorisnika: string
}
